core/src/main/java/org/infinispan/marshall/core/internal/InternalIds.java

```diff
@@ -0,0 +1,14 @@
+package org.infinispan.marshall.core.internal;
+
+final class InternalIds {
+
+   static final int NULL                = 0x00; // null
+   static final int INTERNAL            = 0x01; // internal, including primitives
+   static final int PREDEFINED          = 0x02; // predefined foreign externalizers
+   static final int ANNOTATED           = 0x03; // annotated with @SerializeWith or @SerializeFunctionWith
+   static final int EXTERNAL            = 0x04; // external
```

Comment:
Assuming we can't eliminate these, shouldn't they be in the `MarshallableType` enum?